git-branchless/src/commands/reword.rs

```diff
@@ -528,4 +544,42 @@ This is a template!
 
         Ok(())
     }
+
+    #[test]
+    fn test_parse_bulk_edit_message() -> eyre::Result<()> {
+        let git = make_git()?;
+        git.init_repo()?;
+        let repo = git.get_repo()?;
+
+        let test1_oid = git.commit_file("test1", 1)?;
+        let test2_oid = git.commit_file("test2", 2)?;
+        let test1_commit = repo.find_commit_or_fail(test1_oid)?;
+        let test2_commit = repo.find_commit_or_fail(test2_oid)?;
+
+        {
+            let result = parse_bulk_edit_message(
+                String::from(
+                    "++ reword 62fc20d\n\
+                create test1.txt\n\
+                \n\
+                ++ reword 96d1c37\n\
+                create test2.txt\n",
+                ),
+                &[test1_commit.clone(), test2_commit.clone()],
+                '#',
+            )?;
+            
+            assert_eq!(2, result.messages.len());
```

Comment:
I'm not quite sure I pulled this off quite like you wanted. I pulled `result.messages` out into a `BTreeMap` for sorting, but also had to clear out `result.messages` so that the other elements could be snapshotted w/ the non-deterministic `messages` present. So this does use snapshots, but it uses 2 of them. If you had something else is mind, please let me know.